### PR TITLE
Makes fireball better at breaking things

### PR DESCRIPTION
**Pull request by Sprok0**
When a fireball hits something that isn't a mob, it will now move onto whatever it hits before exploding to increase its effectiveness at breaking things that aren't the floor. No longer will a window take like ten fireballs to break!

Adds a bit to the fireball description to make clearer that it will explode if it gets within a tile of any mobs.
Makes fireball not get stopped by people lying on the ground.

### CODE REVIEW

**Comment by Allspice** [comment]
should move onto mobs too I think, but this is a good start.

**Comment by Sprok0** [comment]
If it moved onto mobs it would be be a one-shot crit every time, that's a bit too much for a ten second recharge spell

**Comment by Allspice** [comment]
Wizards dun hev magboots though, unless they have norobes and go lookin for a pair.
Also fireball wizards still kill themselves near every time, catching internal from light radius will still smart like hell.

**Comment by SonixApache** [comment]
you're a goddamn hero

**Comment by clusterfack** [comment]
How much testing went into this famalam

**Comment by ComicIronic** [comment]
Doesn't this make fireball less deadly at point blank for the wizard?

**Comment by Sprok0** [comment]
100% tested, and it does make fireball not instant death to the wizard if they're facing a wall or something. They still die if they're on the inner side of a window or firelock though, and the light explosion damage can still break bones and cause internal bleeding, so it's not like it's safe for the wizard.

File: code/modules/spells/targeted/projectile/fireball.dm [hunk 1]
```diff
@@ -36,6 +36,13 @@
 	explosion(get_turf(spell_holder), ex_severe, ex_heavy, ex_light, ex_flash)
 	return targets
 
+/spell/targeted/projectile/dumbfire/fireball/choose_prox_targets(mob/user = usr, var/atom/movable/spell_holder)
+	var/list/targets = ..()
+	for(var/mob/living/M in targets)
+		if(M.isUnconscious())
```
Comment (Intigracy) [inline code/modules/spells/targeted/projectile/fireball.dm]: Does this check if they're resting?
Fireballs should pass over people lying down, imo.

**Pull request merged**